dxr/plugins/rust/tests/test_fns/test_fns.py

```diff
@@ -0,0 +1,82 @@
+from dxr.testing import DxrInstanceTestCase
+
+class FunctionTests(DxrInstanceTestCase):
+    def test_simple_function(self):
+        self.found_line_eq('function:foo', "fn <b>foo</b>() {", 3)
+
+    def test_case_insensitive(self):
+        self.found_line_eq('function:FOO', "fn <b>foo</b>() {", 3,
+                           is_case_sensitive=False)
+
+    def test_methods_are_functions_too(self):
+        self.found_line_eq('function:baz', "fn <b>baz</b>(&amp;self) {}", 10)
```

Comment:
It would be nice to comment (or break out into separate tests) to make clear what each of these assertions test: I would guess (1) methods in impls, (2) method declarations (or whatever they're called in Rust) in traits, (3) method definitions in traits.